lib/alexa_request_verifier.rb

```diff
@@ -29,7 +29,7 @@ def valid!(request)
       AlexaRequestVerifier::Verifier::CertificateURIVerifier.valid!(signature_certificate_url)
 
       raw_body = request.body.read
-      request.body.rewind
+      request.body && request.body.rewind # call the rewind method if it exists (handles Sinatra specifically)
```

Comment:
Is the idea here that `request.body.rewind` is called iff `request.body` doesn't get unset after the first read?  (Not super familiar with how Rack handles this).

Reply:
Well, it's not technically as smart as that - this is calling the rewind method on body if the method exists. In newer versions of Ruby >= 2.3 you could use safe navigation which would make this line:

```ruby
request.body&.rewind
```

which is the equivalent of the above - but I was keen to keep the gem backwards compatible with Ruby 2.0+

---

Effectively this line needs to be conditional as inside of Rails the request doesn't have a rewind method